RedfishPkg/Library/RedfishHttpLib/RedfishHttpLib.c

```diff
@@ -0,0 +1,585 @@
+  IN  EFI_STRING        Uri,
+  OUT REDFISH_RESPONSE  *Response
+  )
+{
+  if (mRedfishHttpProtocol == NULL) {
+    return EFI_NOT_READY;
+  }
+
+  return mRedfishHttpProtocol->DeleteResource (
+                                 mRedfishHttpProtocol,
+                                 Service,
+                                 Uri,
+                                 NULL,
+                                 0,
+                                 NULL,
+                                 Response
+                                 );
+}
+
+/**
+  Perform HTTP DELETE to delete redfish resource on given resource URI.
+  It's caller's responsibility to free Response by calling RedfishHttpFreeResponse ().
+
+  @param[in]  Service       Redfish service instance to make query.
+  @param[in]  Uri           Target resource URI.
+  @param[in]  Content       JSON represented properties to be deleted. This is
+                            optional.
+  @param[in]  ContentSize   Size of the Content to be send to Redfish service.
+                            This is optional. When ContentSize is 0, ContentSize
+                            is the size of Content if Content is not NULL.
+  @param[in]  ContentType   Type of the Content to be send to Redfish service.
+                            This is optional.
+  @param[out] Response      HTTP response from redfish service.
+
+  @retval     EFI_SUCCESS     Resrouce is returned successfully.
+  @retval     Others          Errors occur.
+
+**/
+EFI_STATUS
+RedfishHttpDeleteResourceEx (
+  IN  REDFISH_SERVICE   Service,
+  IN  EFI_STRING        Uri,
+  IN  CHAR8             *Content OPTIONAL,
+  IN  UINTN             ContentSize OPTIONAL,
+  IN  CHAR8             *ContentType OPTIONAL,
+  OUT REDFISH_RESPONSE  *Response
+  )
+{
+  if (mRedfishHttpProtocol == NULL) {
+    return EFI_NOT_READY;
+  }
+
+  return mRedfishHttpProtocol->DeleteResource (
+                                 mRedfishHttpProtocol,
+                                 Service,
+                                 Uri,
+                                 Content,
+                                 ContentSize,
+                                 ContentType,
+                                 Response
+                                 );
+}
+
+/**
+  Callback function when gEdkIIRedfishHttpProtocolGuid is installed.
+
+  @param[in] Event    Event whose notification function is being invoked.
+  @param[in] Context  Pointer to the notification function's context.
+**/
+VOID
+EFIAPI
+RedfishHttpProtocolIsReady (
+  IN  EFI_EVENT  Event,
+  IN  VOID       *Context
+  )
+{
+  EFI_STATUS  Status;
+
+  if (mRedfishHttpProtocol != NULL) {
+    gBS->CloseEvent (Event);
+    return;
+  }
+
+  Status = gBS->LocateProtocol (
+                  &gEdkIIRedfishHttpProtocolGuid,
+                  NULL,
+                  (VOID **)&mRedfishHttpProtocol
+                  );
+  if (EFI_ERROR (Status)) {
+    return;
+  }
+
+  gBS->CloseEvent (Event);

```

Comment:
Would you like to move this to the beginning of this function? Also remove the one when if (mRedfishHttpProtocol != NULL)?

Reply:
We only close the event when we can locate Redfish HTTP protocol, right?

Reply:
But we should also close the event when we locate HTTP protocol successfully?